idea/src/org/jetbrains/kotlin/idea/inspections/SetterBackingFieldAssignmentInspection.kt

```diff
@@ -35,12 +36,18 @@ class SetterBackingFieldAssignmentInspection : AbstractKotlinInspection(), Clean
                                 arg.text == parameter?.text
                                         && arg.getArgumentExpression().getResolvedCall(context)?.resultingDescriptor == parameterDescriptor
                             }
+                        is KtNameReferenceExpression -> {
+                            hasBackingField = it.text == KtTokens.FIELD_KEYWORD.value
```

Comment:
There is `BindingContext.BACKING_FIELD_REQUIRED`, please use it and not a raw text check